File: UnityVision-iOS/Assets/Scripts/Possible/Vision/Plugins/iOS/VisionClassification.swift
//
//  VisionClassification.swift
//  Vision
//
//  Created by Adam Hegedus on 2018. 06. 15..
//

import Foundation

@objc public class VisionClassification : NSObject
{
    @objc public var identifier : String = ""
    @objc public var confidence : Float = 0
    
    public init(identifier: String, confidence: Float)
    {
        self.identifier = identifier
        self.confidence = confidence
    }
}

File: UnityVision-iOS/Assets/Scripts/Possible/Vision/Managed/Bridging/VisionClassification.cs
///////////////////////////////////////////////////////////////////////////////
// VisionClassification.cs
// 
// Author: Adam Hegedus
// Contact: adam.hegedus@possible.com
// Copyright © 2018 POSSIBLE CEE. Released under the MIT license.
///////////////////////////////////////////////////////////////////////////////

using System.Runtime.InteropServices;

namespace Possible.Vision.Managed.Bridging
{
    /// <summary>
    /// Represents a single entry extracted from the native classification result buffer. 
    /// </summary>
    [StructLayout(LayoutKind.Sequential)]
    public struct VisionClassification
    {
        public float confidence;
        public string identifier;
    }
}